Libs/MRML/Core/vtkMRMLSegmentationDisplayNode.cxx

```diff
@@ -365,62 +368,121 @@ void vtkMRMLSegmentationDisplayNode::SetSegmentDisplayProperties(std::string seg
       }
     }
 
-  // Save cached value of color
-  // TODO: remove this when terminology infrastructure is in place
+  if (modified)
+    {
+    this->Modified();
+    }
+}
+
+//---------------------------------------------------------------------------
+vtkVector3d vtkMRMLSegmentationDisplayNode::GetSegmentColor(std::string segmentID)
+{
+  double color[3] = {vtkSegment::SEGMENT_COLOR_INVALID[0], vtkSegment::SEGMENT_COLOR_INVALID[1], vtkSegment::SEGMENT_COLOR_INVALID[2]};
+  this->GetSegmentColor(segmentID, color);
+
+  vtkVector3d colorVtk(color[0], color[1], color[2]);
+  return colorVtk;
+}
+
+//---------------------------------------------------------------------------
+void vtkMRMLSegmentationDisplayNode::GetSegmentColor(std::string segmentID, double* color)
+{
+  if (!color)
+    {
+    vtkErrorMacro("GetSegmentColor: Invalid output color array");
+    return;
+    }
+
+  // Invalidate color
```

Comment:
It would better to not constrain developers to set segmentation node before they can set/get override color. We should probably check first if override color is set for a specific segment ID and if it is set then return the override color. If override color is not found then check the color in segmentation node.